src/bucket/media.js

```diff
@@ -84,12 +85,8 @@ const mediaChainMethods = (bucket_config) => ({
     }
     return requestHandler(HTTP_METHODS.DELETE, endpoint, null, headers)
   },
-  async then(resolve) {
-    resolve(
-      new Promise((res) => {
-        res(requestHandler(HTTP_METHODS.GET, this.endpoint))
-      })
-    )
+  async then(cb) {
+    promiser(this.endpoint).then((res) => cb(res))
```

Comment:
Looking great! I think only thing that is missing is error handling. I'm not sure if it will throw the errors properly. Maybe, we check that case as well. 

Possible solution!!

```
promiser(this.endpoint).then((res) => cb(res, null)).catch((err) => cb(null, err));
```